File: #19.Remove Nth Node From End of List/Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
 /*
 自己添加头结点
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *front,*rear;
        ListNode addHead(0);    //添加自己的头结点，方便处理节点
        addHead.next=head;  
        front=&addHead;
        --n;    //实际寻找倒数第n-1个节点
        while(n>0){
            front=front->next;
            --n;
        }
        rear=&addHead;
        while(front->next->next!=NULL){
            front=front->next;
            rear=rear->next;
        }
        rear->next=rear->next->next;//删除rear下一个节点
        head=addHead.next;
        return head;
        
        
    }
};

File: #19.Remove Nth Node From End of List/Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {

    if(head == null || n < 0){
        return head;
    }
    var addHead=new ListNode(-1);
    addHead.next=head;

    var front=addHead;
    var rear=addHead;

    while(n>0){
        front=front.next;
        --n;
    }

    while(front.next!=null){
        front=front.next;
        rear=rear.next;
    }

    rear.next=rear.next.next;

    return addHead.next;

};